Crashing when trying to add cache to watchlist

###### Detailed steps causing the problem:
- go to cache details 
- press button "add to watchlist"

###### Actual behaviour after performing these steps:
C:geo restarts and return to home page 
Cache is not added to watchlist 

###### Expected behaviour after performing these steps:
Cache should be added to watchlist 


###### Is the problem reproducible for you:
Yes


###### Attach system information if available
--- System information ---
Device: ONE E1003 (OnePlus, OnePlus)
Android version: 5.1.1
Android build: ONE E1003_14_160122
Cgeo version: 2016.02.23-NB-891182e
Google Play services: enabled
Low power mode: inactive
Compass capabilities: yes
Rotation vector sensor: present
Orientation sensor: present
Magnetometer & Accelerometer sensor: present
Direction sensor used: rotation vector
Hide own/found: false
Map strategy: auto
HW acceleration: enabled (default state)
System language: sv_SE
Geocaching sites enabled:
 - geocaching.com: logged in (Inloggning OK) / PREMIUM
Installed cgeo plugins: calendar
--- End of system information ---




###### Other comments and remarks:
Noticed the bug in 2016.02.22 NB

CAn reproduce this on with the same version on Android 4.4.4.

Log:
```
22:36:41.993 Error cgeo 16923  [network-10] UncaughtException
22:36:41.994 Error cgeo 16923  java.lang.IllegalStateException: Fatal Exception thrown on Scheduler.Worker thread.
22:36:41.994 Error cgeo 16923  	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:62)
22:36:41.994 Error cgeo 16923  	at rx.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:98)
22:36:41.994 Error cgeo 16923  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
22:36:41.994 Error cgeo 16923  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
22:36:41.994 Error cgeo 16923  	at java.lang.Thread.run(Thread.java:841)
22:36:41.994 Error cgeo 16923  Caused by: java.lang.IllegalArgumentException: method POST must have a request body.
22:36:41.994 Error cgeo 16923  	at okhttp3.Request$Builder.method(Request.java:239)
22:36:41.994 Error cgeo 16923  	at okhttp3.Request$Builder.post(Request.java:212)
22:36:41.994 Error cgeo 16923  	at cgeo.geocaching.network.Network.request(Network.java:176)
22:36:41.994 Error cgeo 16923  	at cgeo.geocaching.network.Network.postRequest(Network.java:86)
22:36:41.994 Error cgeo 16923  	at cgeo.geocaching.connector.gc.GCLogin.postRequestLogged(GCLogin.java:459)
22:36:41.995 Error cgeo 16923  	at cgeo.geocaching.connector.gc.GCParser.addToWatchlist(GCParser.java:1385)
22:36:41.995 Error cgeo 16923  	at cgeo.geocaching.connector.gc.GCConnector.addToWatchlist(GCConnector.java:220)
22:36:41.995 Error cgeo 16923  	at cgeo.geocaching.CacheDetailActivity$DetailsViewCreator.watchListAdd(CacheDetailActivity.java:1188)
22:36:41.995 Error cgeo 16923  	at cgeo.geocaching.CacheDetailActivity$DetailsViewCreator.access$3300(CacheDetailActivity.java:910)
22:36:41.995 Error cgeo 16923  	at cgeo.geocaching.CacheDetailActivity$DetailsViewCreator$AddToWatchlistClickListener$1.call(CacheDetailActivity.java:1163)
22:36:41.995 Error cgeo 16923  	at cgeo.geocaching.CacheDetailActivity$DetailsViewCreator$AddToWatchlistClickListener$1.call(CacheDetailActivity.java:1160)
22:36:41.995 Error cgeo 16923  	at cgeo.geocaching.CacheDetailActivity$DetailsViewCreator$AbstractPropertyListener$2.call(CacheDetailActivity.java:1146)
22:36:41.995 Error cgeo 16923  	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
22:36:41.995 Error cgeo 16923  	... 4 more
```

--- a/main/src/cgeo/geocaching/network/Network.java
+++ b/main/src/cgeo/geocaching/network/Network.java
@@ -164,15 +164,13 @@ public final class Network {
             builder.url(urlBuilder.build());
         } else {
             builder.url(uri);
+            final FormBody.Builder body = new FormBody.Builder();
             if (params != null) {
-                final FormBody.Builder body = new FormBody.Builder();
-                for (final ImmutablePair<String, String> param: params) {
+                for (final ImmutablePair<String, String> param : params) {
                     body.add(param.left, param.right);
                 }
-                builder.post(body.build());
-            } else {
-                builder.post(null);
             }
+            builder.post(body.build());
         }
 
         addHeaders(builder, headers, cacheFile);

